source/api/content-state/0.9/index.md

```diff
@@ -725,37 +602,185 @@ Firstly, in non-valid, unencoded form to show the annotation:
+* remove any "=" padding characters from the end
+
+Conversely to decode a content state:
+
+* restore any removed "=" padding characters to the end of the string to make it a multiple of 4 characters long.
+* decode the resulting string from base64url to a UTF-8 string
+* decode the resulting UTF-8 string as described by [decodeURIComponent][org-ecma-decodeuricomponent]
+
+Code samples for these operations are given in the next section.
+
+Any content state that is in JSON-LD form, rather than a simple URI string, _MUST_ be _content-state-encoded_ in this way when passed as a GET parameter on a query string, and a client _MUST_ accept it in this form.
+
+Simple URI forms _MAY_ be either plain strings, URI-encoded plain strings, or content-state-encoded strings as above, and a client _MUST_ accept these forms and distinguish between them.
+
+Any content state passed by mechanisms other than a HTTP GET request parameter _MUST NOT_ be content-state-encoded.
+
+When published as inline, encoded JSON-LD in the full form given in section 2.2. above, the content state Annotation _MAY_ omit the `id` and `@context` properties.
+
+When published on a server for clients to fetch over HTTP, in the same way a client would fetch a Manifest or Collection, content states _MUST_ be valid JSON-LD documents conforming to the [IIIF Presentation API][prezi-api] and served as described in [Section 6][contentstate-http] below. They _MUST NOT_ be content-state-encoded, but _MAY_ have other encodings appropriate for JSON content, such as `Content-Encoding: gzip` to reduce the response size.
+
+
+### 6.2. Examples of Content State Encoding
+
+JavaScript and Python examples are given below. Examples for other languages and frameworks can be found in the [IIIF Cookbook][annex-cookbook].
+
+#### 6.2.1. JavaScript
+
+```javascript
+function encodeContentState(plainContentState) {
+    let uriEncoded = encodeURIComponent(plainContentState);  // using built in function
+    let base64 = btoa(uriEncoded);                           // using built in function
+    let base64url = base64.replace(/\+/g, "-").replace(/\//g, "_");
+    let base64urlNoPadding = base64url.replace(/=/g, "");
+    return base64urlNoPadding;
+}
+
+
+function decodeContentState(encodedContentState) {
+    let base64url = restorePadding(encodedContentState);
+    let base64 = base64url.replace(/-/g, '+').replace(/_/g, '/');
+    let base64Decoded = atob(base64);                        // using built in function
+    let uriDecoded = decodeURIComponent(base64Decoded);      // using built in function
+    return uriDecoded;
+}
+
+
+function restorePadding(s) {
+    // string length must be a multiple of 4
+    let pad = s.length % 4;
+    if (pad) {
+        if (pad === 1) {
+            throw new Error('InvalidLengthError: Input base64url string is the wrong length to determine padding');
+        }
+        s += new Array(5 - pad).join('=');
```

Comment:
I don't think it does directly. Since `join()` will only apply _between_ elements. So an Array with 5 elements will have 4 "joins".

I would recommend changing to:
```
'===='.slice(0, 4 - pad)
```

For clarity.

Reply:
Yes, my bad! But I agree the the slide method is nicer